### PR TITLE
M3-3841 Fix: Images events messaging & assorted bugs

### PR DESCRIPTION
**Pull request by Jskobos**
## Description

Started out updating Images messaging to account for new secondary_entity capability, and found ~two~ four bugs while I was at it:

- Images list wasn't updating after editing label etc.
- CreateLinodeContainer wasn't looking at the imageID param, so "Deploy image to Linode" action was essentially broken
- ImagesDrawer didn't have a loading state
- Changing selected disk/Linode didn't reset the error state, so you could:
    1. Open the Images drawer
    2. Block requests
    3. Select a Linode
    4. Observe: you get a disks error
    5. Unblock requests and select a new Linode
    6. Observe: error is persistent, can't be gotten rid of until you submit the form.

Refactored a few things too, but there's still more tech debt in there, like many of our older components.

## Type of Change
- Bug fix ('fix', 'repair', 'bug')
- Non breaking change ('update', 'change')

## Note to Reviewers

Make, fail, delete, edit, etc. a bunch of Images. Check the messaging in the events menu, activity summary, and toasts.

### CODE REVIEW

File: packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeDisks.tsx [hunk 1]
```diff
@@ -360,10 +360,14 @@ class LinodeDisks extends React.Component<CombinedProps, State> {
         description={description}
         label={label}
         disks={disk ? [disk] : []}
-        selectedDisk={disk && '' + disk.id}
+        selectedDisk={disk ? '' + disk.id : null}
         onClose={this.closeImagizeDrawer}
         changeDescription={this.changeImageDescription}
         changeLabel={this.changeImageLabel}
+        changeDisk={() => null}
```
Comment (Jskobos) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeDisks.tsx]: We could make these optional, but this is only one place and I'd rather the awkwardness live here than add conditional logic for all the handlers in the drawer.

File: packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeDisks.tsx [hunk 2]
```diff
@@ -360,10 +360,14 @@ class LinodeDisks extends React.Component<CombinedProps, State> {
...
+        onSuccess={() => null}
```
Comment (johnwcallahan) [inline packages/manager/src/features/linodes/LinodesDetail/LinodeAdvanced/LinodeDisks.tsx]: It doesn't look like any parent of `<ImagesDrawer />` gives it an `onSuccess` prop... we might be able to remove it.

File: packages/manager/src/eventMessageGenerator.ts [hunk 3]
```diff
@@ -165,14 +165,14 @@ export const eventMessageCreators: { [index: string]: CreatorsForStatus } = {
       `A domain record has been deleted from ${e.entity!.label}`
   },
   image_update: {
-    notification: e => `Image ${e.entity!.label} has been updated.`
+    notification: e => `Image ${e.entity?.label ?? ''} has been updated.`
```
Comment (acourdavault) [inline packages/manager/src/eventMessageGenerator.ts]: `Image ${e.entity?.label ?? ''} has been updated.` =>`Image ${e.entity?.label ?? ''}has been updated.`

File: packages/manager/src/eventMessageGenerator.ts [hunk 4]
```diff
@@ -165,14 +165,14 @@ export const eventMessageCreators: { [index: string]: CreatorsForStatus } = {
...
-    // scheduled: e => `Image ${e.entity!.label} scheduled for deletion.`,
-    // started: e => `Image ${e.entity!.label} is being deleted.`,
-    // failed: e => `There was a problem deleting ${e.entity!.label}.`,
-    // finished: e => `${e.entity!.label}`,
-    notification: e => `Image ${e.entity!.label} has been deleted.`
+    scheduled: e => `Image ${e.entity?.label ?? ''} scheduled for deletion.`,
+    started: e => `Image ${e.entity?.label ?? ''} is being deleted.`,
+    failed: e => `There was a problem deleting ${e.entity?.label ?? ''}.`,
+    finished: e => `Image ${e.entity?.label ?? ''} has been deleted.`,
+    notification: e => `Image ${e.entity?.label ?? ''} has been deleted.`
```
Comment (acourdavault) [inline packages/manager/src/eventMessageGenerator.ts]: same thing with the extra space

File: packages/manager/src/features/ToastNotifications/ToastNotifications.tsx [hunk 5]
```diff
@@ -54,33 +54,47 @@ class ToastNotifications extends React.PureComponent<WithSnackbarProps, {}> {
...
         if (
           event.action === 'disk_imagize' &&
           ['finished', 'notification'].includes(event.status)
         ) {
-          return enqueueSnackbar('Image created successfully.', {
-            variant: 'success'
-          });
+          return enqueueSnackbar(
+            `Image ${event.secondary_entity?.label ??
+              ''} created successfully.`,
```
Comment (acourdavault) [inline packages/manager/src/features/ToastNotifications/ToastNotifications.tsx]: same extra space

File: packages/manager/src/features/Images/ImagesDrawer.tsx [hunk 6]
```diff
@@ -87,7 +93,8 @@ class ImageDrawer extends React.Component<CombinedProps, State> {
   state = {
     disks: [],
     errors: undefined,
-    notice: undefined
+    notice: undefined,
```
Comment (acourdavault) [inline packages/manager/src/features/Images/ImagesDrawer.tsx]: Here, not a problem, but out of curiosity why do we init with undefined and not null ?

Comment (Jskobos) [reply packages/manager/src/features/Images/ImagesDrawer.tsx]: convention, basically. null would probably be better.

**Pull request merged**